src/main/java/frc/robot/commands/SequentialAutoCommand.java

```diff
@@ -1,217 +1,104 @@
-                                m_drivetrain),
-                        // Rotate towards hub
-                        new Rotate(m_drivetrain, 180),
-                        // Drive to launching position
-                        new DriveTo(new Position2D(10.412, 7.359, Math.toRadians(0)), 2.0d, false, m_kinematics,
-                                m_drivetrain),
-                        // Launch terminal ball
-                        new ParallelRaceGroup(
-                                new Wait(3.0)));
-                break;
-
-            case RED_RIGHT:
-                addCommands(
-                        // Reset kinematics to red right position
-                        new ResetKinematics(new Position2D(1.995, 7.628, Math.toRadians(-91.5)), m_drivetrain,
-                                m_kinematics),
-                        // Drives to first ball and collects
-                        new DriveTo(new Position2D(2.159, 12.566, Math.toRadians(0)), 2.0d, false, m_kinematics,
-                                m_drivetrain),
-                        // Rotate to face the hub
-                        new Rotate(m_drivetrain, 180.0),
-                        // Aim at the hub
-                        new TrackTarget(m_drivetrain, m_targeting),
-                        // Fires both balls!
-                        new ParallelRaceGroup(
-                                new Wait(3.0)),
-                        // Rotates towards opposing ball
-                        new Rotate(m_drivetrain, -90.0),
-                        // drives to opposing ball
-                        new DriveTo(new Position2D(-2.814, 12.436, Math.toRadians(0)), 2.0d, false, m_kinematics,
-                                m_drivetrain),
-                        // Rotate away from hub
-                        new Rotate(m_drivetrain, 180.0),
-                        // Dumps enemy ball
-                        new ParallelRaceGroup(
-                                new Wait(3.0)));
-                break;
-
-            default:
-                System.out.println("ERROR: Invalid autonomous starting position! [" + m_startPosition + "]");
-                break;
-        }
-
-        // Alert smart dashboard that autonomous is done
-        SmartDashboard.putBoolean("AutoDone", true);
-    }
+	private StartPositions m_startPosition;
+	private Drivetrain m_drivetrain;
+	private Arm m_arm;
+	private Claw m_claw;
+	private Kinematics m_kinematics;
+
+	public SequentialAutoCommand(Drivetrain drivetrain, Arm arm, Claw claw, Kinematics kinematics, Targeting targeting,
+			StartPositions startPosition) {
+		m_drivetrain = drivetrain;
+		m_arm = arm;
+		m_claw = claw;
+		m_kinematics = kinematics;
+		m_startPosition = startPosition;
+
+		SmartDashboard.putBoolean("AutoDone", false);
+
+		switch (m_startPosition) { // Changes the robot path based on the starting position of the robot Left,
+			// Middle, Right
+			case BLUE_LEFT:
+				bottom();
```

Comment:
Don't split these into individual functions. Just create the commands here.

Reply:
Because some autos are duplicates, they are split into individual methods. This way, if we chance the bottom method, it will change for both blue and red.